lib/wasix/src/state/linker.rs

```diff
@@ -1351,16 +1496,109 @@ impl Linker {
         }
     }
 
+    /// TODO: Arshia told me not to do this but I want to implement the other stuff first.
```

Comment:
Love the honorable mention 😄 indeed, this will break when threads are involved.

Reply:
I am not really sure where to move the function table access, if not into the linker, because the function table isn't an export of a module but is stored in the instance group. I could move a copy outside early on. Do you have any suggestions on where to put it?

I at least tried to mitigate it by acquiring the instance group lock. Can you take another look?